image/templates/helm/shared/templates/_scanner-v4_init.tpl.htpl

```diff
@@ -24,150 +24,114 @@
-    {{- $_ = set $components "matcher" (not $scannerV4Cfg.matcher.disable) -}}
-  {{- end -}}
 
-  {{/* This is potentially an unsupported configuration. Hence we generate a warning. */}}
-  {{- $msg := printf "indexer is %s and matcher is %s." (ternary "enabled" "disabled" (get $components "indexer")) (ternary "enabled" "disabled" (get $components "matcher")) -}}
-  {{- if eq $.Chart.Name "stackrox-central-services" -}}
-    {{- if or (ne (get $components "indexer") (get $componentsCentralChart "indexer")) (ne (get $components "matcher") (get $componentsCentralChart "matcher")) -}}
-      {{- include "srox.warn" (list $ (printf "Unsupported Scanner V4 configuration detected: %s" $msg)) -}}
-    {{- end -}}
-  {{- else -}}
-    {{- if or (ne (get $components "indexer") (get $componentsSecuredClusterChart "indexer")) (ne (get $components "matcher") (get $componentsSecuredClusterChart "matcher")) -}}
-      {{- include "srox.warn" (list $ (printf "Unsupported Scanner V4 configuration detected: %s" $msg)) -}}
+  {{/* Configure images, certificates and passwords as required. */}}
+  {{ if get $components "indexer" }}
+    {{/* Note that for secured-cluster-services we don't configure certificates here,
+         instead they will be distributed at runtime by Sensor and Central. */}}
+    {{ include "srox.configureImage" (list $ $scannerV4Cfg.indexer.image) }}
+    {{- if eq $.Chart.Name "stackrox-central-services" -}}
+      {{/* Only generate certificate when installing central-services. */}}
+      {{- if kindIs "invalid" $._rox.scannerV4.indexer.serviceTLS.generate }}
+        {{/* We need special handling here, because 'generate' will default to 'false' on upgrades. */}}
+        {{/* And in case scanner V4 was not deployed earlier, we need to make sure that these resources are correctly initialized. */}}
+        {{- if $.Release.IsUpgrade -}}
+          {{- $lookupOut := dict -}}
+          {{- $_ := include "srox.safeLookup" (list $ $lookupOut "v1" "Secret" $.Release.Namespace "scanner-v4-indexer-tls") -}}
+          {{- if not $lookupOut.result -}}
+            {{/* If generate=null and the resource does not exist yet, attempt to create it. */}}
+            {{/* If lookup is not possible (e.g. in the operator), then 'generate' needs to be set correctly. */}}
+            {{- $_ := set $._rox.scannerV4.indexer.serviceTLS "generate" true -}}
+          {{- end -}}
+        {{- end -}}
+      {{- end }}
+      {{ $cryptoSpec := dict "CN" "SCANNER_V4_INDEXER_SERVICE: Scanner V4 Indexer" "dnsBase" "scanner-v4-indexer" }}
+      {{ include "srox.configureCrypto" (list $ "scannerV4.indexer.serviceTLS" $cryptoSpec) }}
     {{- end -}}
-  {{- end -}}
+  {{ end }}
 
-{{- end -}}
+  {{ if get $components "matcher" }}
+    {{/* This is only valid when rendering the central-services chart. */}}
+    {{ include "srox.configureImage" (list $ $scannerV4Cfg.matcher.image) }}
+    {{- if eq $.Chart.Name "stackrox-central-services" -}}
```

Comment:
if we are sure we are only deploying in central services, do we still need this check? maybe for extra safety?

Reply:
Hmm, yeah, I will get rid of it.